### ISSUE TITLE
Is there a way to generate a confusion matrix?

### ISSUE THREAD

**Issue by raulmat19** [sovit-123/fasterrcnn-pytorch-training-pipeline#72]
Hi! I would love to know if there is a way or could be implemented a way of generating a confusion matrix while validation, among other metrics as F1-score.

**Comment by sovit-123** [comment]
Hello @raulmat19 
Most probably it should originate from the `evaluate()` function in the `engine.py` file. But I will have to take a closer look to implement it in a clean way.

**Comment by raulmat19** [comment]
Hello! Okey, thanks man. By the way, it would also be interesting to use the evaluate() function to show some validation loss graphs.

**Comment by sovit-123** [comment]
@raulmat19 
I completely understand your concern. It is a bit tricky to do with Torchvision object detection models. But I will try again for that in a few days.

**Comment by Engineer-D** [comment]
I tried something I don't know if it makes any sense to you.

So in the eval.py using the length of the true prediction I was able to cut down my predicted predictions to be exactly the length of the true prediction (if my predicted is less than my True I append Zeros as predicted as a background). I then appended all this values to an external list for each of predicted and actual values.

The idea here is that all classes from all true and predicted images have been combined in a list. then I did confusion matrix on that using torchmetrics.classification

**Comment by Engineer-D** [comment]
yes sure

**Comment by Engineer-D** [comment]
would you mind if I write the code and push it to you?

**Comment by sovit-123** [comment]
If you can create a PR that would be great. I pushed some changes yesterday. But they were to `train.py` and `datasets.py`. Basically, mosaic augmentation was applied all the time or not applied based on one command line argument. Now, I made it a float value so that it can be passed a probability between 0 and 1. Please check once whether you need to pull the new code.

**Comment by raulmat19** [comment]
Hey @Engineer-D! I wanted to thank you, that looks good. If you could create the PR or paste here the code would be great.

**Comment by mtalhaubaid** [comment]
@Engineer-D have you pushed the code this?
